### PR TITLE
Remove synchronized block in Thread isDead method

### PR DESCRIPTION
**Pull request by ThanHenderson**
Related: https://github.com/eclipse-openj9/openj9/pull/20415
Related: https://github.com/eclipse-openj9/openj9/issues/20414

Signed-off-by: Nathan Henderson <nathan.henderson@ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
jenkins test sanity.functional alinux64 jdknext

File: src/java.base/share/classes/java/lang/Thread.java [hunk 1]
```diff
@@ -3149,10 +3149,8 @@ private Thread(String vmName, Object vmThreadGroup, int vmPriority, boolean vmIs
     }
 
     private boolean isDead() {
-        // Has already started, is not alive anymore, and has been removed from the ThreadGroup
-        synchronized (interruptLock) {
-            return (started && (eetop == NO_REF));
-        }
+        /* Has already started and is not alive anymore. */
+        return (started && (eetop == NO_REF));
```
Comment (keithc-ca) [inline src/java.base/share/classes/java/lang/Thread.java:3153]: Please remove the outermost parentheses:
```c
        return started && (eetop == NO_REF);
```

**Comment by ThanHenderson** [comment]
PR builds passed:
https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_sanity.functional_aarch64_linux_Personal/50/

Just going to push Keith's requested change which will not require new PR builds.

**Review by babsingh — APPROVED** [review]

**Comment by babsingh** [comment]
Merging. All feedback has been addressed. @keithc-ca's away for the remaining week.

**Pull request merged**